h/admin/services/user.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h import models
+from h.api.search.index import BatchIndexer
+from h.util.user import userid_from_username
+
+
+class UserRenameError(Exception):
+    pass
+
+
+class RenameUserService(object):
```

Comment:
The main reason why I went for a `RenameUserService` rather than a `UserAdminService` was that I felt that the `rename_user` code is complex enough to warrant being split out into a bunch of methods for better readability.

I don't think there's a need right now to move the `delete_user` function somewhere else (given this is work wasn't planned to happen and we are quite busy), but once we are doing that, then we should think about it again and maybe move some stuff around. What do you think?